tests/bip32/test_bip32.py

```diff
@@ -362,3 +363,17 @@ def test_bips_pr905() -> None:
     assert derive(xroot, der_path) == xprv
     xpub = "xpub6CpsfWjghR6XdCB8yDq7jQRpRKEDP2LT3ZRUgURF9g5xevB7YoTpogkFRqq5nQtVSN8YCMZo2CD8u4zCaxRv85ctCWmzEi9gQ5DBhBFaTNo"
     assert xpub_from_xprv(xprv) == xpub
+
+
+def test_pub_key_derivation() -> None:
+    parent_xpub = "xpub6CpsfWjghR6XdCB8yDq7jQRpRKEDP2LT3ZRUgURF9g5xevB7YoTpogkFRqq5nQtVSN8YCMZo2CD8u4zCaxRv85ctCWmzEi9gQ5DBhBFaTNo"
+    proper_child = "xpub6FCCuDg6j52SWRVZ1TugkjrnGkqPcDuNNKDzohU2mmd4dxiGJypZa535iqYT8KcN2oouRF7A6tXEGAX6HCSjQe7HVSDR4LQ4yUT3HwF1Tqi"
+    assert derive(parent_xpub, "m/0") == proper_child
+    parent_key = BIP32KeyData.b58decode(parent_xpub).key
+    parent_fingerprint = hash160(parent_key)[:4]
+    assert BIP32KeyData.b58decode(proper_child).parent_fingerprint == parent_fingerprint
+
+    orphan_child_key = BIP32KeyData.b58decode(proper_child)
+    orphan_child_key.parent_fingerprint = b"\x00" * 4
+    orphan_child = "xpub6DXuQW1FgeHbhsSchbuDWE9Bj8mPiPUpiroAmAvRdRqYbGHXHTyEkttkxSvtCac64QzpasL1Tvd5Znvn5GQMswQUrpRBsPRz7npvyZ8ExWi"
+    assert orphan_child_key.b58encode() == orphan_child
```

Comment:
Where does this test come from?

Reply:
from nowhere, it has been added to test proper public_key_derivation. All BIP32 test vectors always provide private key, so one could test private_key_derivation and neutering (priv->pub), avoiding any test for public_key_derivation. This happened to btclib test-suite. I am going to propose this new test to BIP32 to mitigate this implementation risk